example/kokkos_pi/kokkos_pi.cpp

```diff
@@ -0,0 +1,299 @@
+    View2D xyz("xyzBlocks", 3, n_mesh3);
+    auto h_xyz = Kokkos::create_mirror_view(xyz);
+
+    // *** Kludge warning ***
+    // Since our mesh is not GPU friendly we set up a hacked up
+    // collection of mesh blocks.  The hope is that when our mesh is
+    // up to par we will replace this code with the mesh
+    // infrastructure.
+
+    // Set up our mesh.
+    Metadata myMetadata({Metadata::Independent, Metadata::Cell});
+
+    MeshBlock *lastBlock = nullptr;
+    MeshBlock *allBlocks[n_mesh3]; // probably not needed, but creating it so that we can
+                                   // launch many blocks
+
+    std::cout << "Begin setup " << std::endl;
+    int idxMesh = 0; // an index into AllBlocks array
+    for (int k_mesh = 0; k_mesh < n_mesh; k_mesh++) {
+      for (int j_mesh = 0; j_mesh < n_mesh; j_mesh++) {
+        for (int i_mesh = 0; i_mesh < n_mesh; i_mesh++, idxMesh++) {
+          // get a new meshblock
+          auto *pmb = new MeshBlock(n_block, 3);
+          // if not first block then add to chain
+          if (lastBlock) {
+            // set pointers accordingly
+            lastBlock->next = pmb;
+            pmb->prev = lastBlock;
+          }
+
+          // store block away
+          lastBlock = allBlocks[idxMesh] = pmb;
+          // set coordinates of first cell center
+          h_xyz(0, idxMesh) = dxyzCell * (static_cast<Real>(i_mesh * n_block) + 0.5);
+          h_xyz(1, idxMesh) = dxyzCell * (static_cast<Real>(j_mesh * n_block) + 0.5);
+          h_xyz(2, idxMesh) = dxyzCell * (static_cast<Real>(k_mesh * n_block) + 0.5);
+          // Add our variable for in_or_out
+          Container<Real> &base = pmb->real_containers.Get();
+          base.setBlock(pmb);
+          base.Add("in_or_out", myMetadata);
+        }
+      }
+    }
+    // copy our coordinates over to Device and wait for completion
+    Kokkos::deep_copy(xyz, h_xyz);
+    Kokkos::fence();
+
+    // first let us do a stupid loop over the mesh
+    // This policy is over entire mesh
+
+    // This policy is over entire mesh
+    auto policy = Kokkos::RangePolicy<>(Kokkos::DefaultExecutionSpace(), 0,
+                                        n_block3 * n_mesh3, Kokkos::ChunkSize(512));
+
+    // This policy is over one block
+    auto policyBlock = Kokkos::RangePolicy<>(Kokkos::DefaultExecutionSpace(), 0, n_block3,
+                                             Kokkos::ChunkSize(512));
+
+    std::cout << "Begin basic timing " << std::endl;
+    double time_basic;
+    {
+      MeshBlock *pStart = allBlocks[0];
+      time_basic = kernel_timer_wrapper(0, n_iter, [&]() {
+        MeshBlock *pmb = pStart;
+        for (int iMesh = 0; iMesh < n_mesh3; iMesh++, pmb = pmb->next) {
+          Container<Real> &base = pmb->real_containers.Get();
+          auto inOrOut = base.PackVariables({Metadata::Independent});
+          // iops = 8  fops = 11
+          Kokkos::parallel_for(
```

Comment:
Because you're launching a kernel per meshblock, this is going to be incredibly slow. It'll be interesting to watch this improve as we implement streams, etc, to hide kernel launch latency.

Reply:
Actually, we might be able to coalesce over blocks, since you have the blocks in a kokkos view. It would be interesting to try that and look at the timing.